### DISCUSSION TITLE
How to find the Strimzi deployed successfuly

### DISCUSSION THREAD

**Discussion by srinivasev** [strimzi/strimzi-kafka-operator#7330] [Q&A]
Hi,

Currently we are deploying strimzi operator chart in our product. Helm tool is used for installing Strimzi in Kubernetes. This is part of some umbrella chart and helm doesn't have the hold on kafka clusters, zookeper clusters (I can say pods) whether it is installed or not. In the top level, helm is saying the chart is installed. In the background, these pods are still coming up. In some cases, we also seen the pods are in error state, but the helm status say's it is deployed.

Is there way Strimzi provides to check the deployment status. Not only the helm deployment, it should consider all its custom resources which are tied up with it.

In simple terms, lets say I can run some kubectl command on the cluster and see whether strimzi chart is deployed successfully or not.

**Comment by scholzj** [comment]
I'm not entirely sure about what part are you talking about as you seem to mix the Strimzi Helm Chart (i.e. just the operator deployment) and the custom resources.

----

For custom resources, they have all the `conditions` section in their `.status`. You can check them for being `Ready`. For example the following command:

```
kubectl wait kafka/my-cluster --for=condition=Ready --timeout=300s -n kafka 
```

Will wait up to 300 seconds for the Kafka cluster to get ready. And if it is not ready after the 300s, it will return failure. And it returns success whenever it is ready.

----

For the operator deployment, IMHO you would need to track the Deployment readiness / Pod readiness. No idea if Helm has something special for it.
